### ISSUE TITLE
Minio server generates empty body notification when object put by java client

### ISSUE THREAD

**Issue by bambulin** [minio/minio-java#818]
Assume simple notification listener:
```
public class S3NotificationListener implements BucketEventListener {
    @Override
    public void updateEvent(NotificationInfo info) {
        logger.log(Level.INFO, "Thread " + Thread.currentThread() + ": S3 notification received");
    }
}

MinioClient minioClient = new MinioClient("http://locahost:9000", "minio", "minio123");
client.listenBucketNotification("bucket", "", "", new String[] {"s3:ObjectCreated:*"}, new S3NotificationListener(resource));
```
put object from another application:
```
MinioClient minioClient = new MinioClient("http://locahost:9000", "minio", "minio123");
client.putObject("bucket", "dir-in-bucket/filename", "path/to/local/file", null, null, null, "application/octet-stream");
```
verify (using minio web interface) the object was uploaded to the server, but the `S3NotificationListener.updateEvent` was not invoked.
Using debugger you can see the response was received, but at https://github.com/minio/minio-java/blob/79f2eabb519182107f23ed4be09a17131b2e3cec/api/src/main/java/io/minio/MinioClient.java#L5741 the body is empty so the notification is discarded and the listener method never invoked.
But if you upload the file manually using minio web interface, the notification is generated correctly and the `S3NotificationListener.updateEvent` is invoked as expected.

minio java version: 6.0.11
minio server docker-compose: 
```
version: '3.1'
services:
  minio1:
    image: minio/minio:RELEASE.2019-10-12T01-39-57Z
    volumes:
      - data1-1:/data1
      - data1-2:/data2
    ports:
      - "9000:9000"
    environment:
      MINIO_ACCESS_KEY: minio
      MINIO_SECRET_KEY: minio123
    command: server /data
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3
volumes:
  data1-1:
  data1-2:
```

**Comment by sinhaashish** [comment]
Tried to replicate the issue but its working as intended.  Did the below steps
bucket : images 

event list 
```
ashish@minio:~/code/go/src/github.com/minio/mc$ ./mc event list myminio/images
arn:minio:sqs::1:webhook   s3:ObjectCreated:*   Filter: 
```

Ran the notification code 
```
   */
  public static void main(String[] args)
    throws IOException, NoSuchAlgorithmException, InvalidKeyException, XmlPullParserException {
    try {

      class TestBucketListener implements BucketEventListener {
        @Override
        public void updateEvent(NotificationInfo info) {
          System.out.println(info.records[0].s3.bucket.name + "/"
            + info.records[0].s3.object.key + " has been created");
        }
      }

      minioClient.listenBucketNotification("images", "", "",
        new String[]{"s3:ObjectCreated:*"}, new TestBucketListener());

    } catch (MinioException e) {
      System.out.println("Error occurred: " + e);
    }
  }
}

```

and then upload the objct using minio-java
```
  minioClient.putObject("images", "dir-in-bucket/filename", "/home/ashish/Downloads/auth.txt", null, null, null, "application/octet-stream");
  ```
  
  Got the below in the console of the notification program
  ```
  images/dir-in-bucket%2Ffilename has been created
  ```

**Comment by sinhaashish** [comment]
@bambulin  you can try the new listenBucketNotification method added in the PR https://github.com/minio/minio-java/pull/834

**Comment by harshavardhana** [comment]
Closing this as fixed and released in 7.0.0
